src/main/kotlin/org/rust/cargo/RefreshCargoProjectAction.kt

```diff
@@ -0,0 +1,39 @@
+package org.rust.cargo
+
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.module.Module
+import org.rust.cargo.toolchain.CargoMetadataService
+import org.rust.cargo.toolchain.RustToolchain
+import org.rust.cargo.toolchain.toolchain
+import org.rust.cargo.util.getModules
+import org.rust.cargo.util.getService
+
+class RefreshCargoProjectAction : AnAction() {
+    init {
+        templatePresentation.text = "Refresh Cargo project"
+        templatePresentation.description = "Update Cargo project information and download new dependencies"
+    }
+
+    override fun update(e: AnActionEvent) {
+        if (e.modulesWithToolchains.isEmpty()) {
+            e.presentation.isEnabled = false
+        }
+    }
+
+    override fun actionPerformed(e: AnActionEvent) {
+        val modules = e.modulesWithToolchains
+        if (modules.isEmpty()) return
+        ApplicationManager.getApplication().saveAll()
+        for ((module, toolchain) in modules) {
+            val service = module.getService<CargoMetadataService>()
+            service.updateNow(toolchain)
+        }
+    }
+}
+
+private val AnActionEvent.modulesWithToolchains: List<Pair<Module, RustToolchain>>
```

Comment:
It's not that better imo because `AnActionEvent.project` is nullable. Updated nevertheless.